src/Build/BackEnd/Components/ProjectCache/ProjectCacheService.cs

```diff
@@ -230,7 +230,10 @@ async Task<CacheResult> ProcessCacheRequest(CacheRequest request)
                     await LateInitializePluginForVsWorkaround(request);
                 }
 
-                return await GetCacheResultAsync(cacheRequest.Submission.BuildRequestData);
+                return await GetCacheResultAsync(
+                    new BuildRequestData(
```

Comment:
Do you think it would make sense to check whether the ProjectInstance is set?

Reply:
Good catch, added it.